### PR TITLE
Refactor executions endpoints for user management

### PR DESCRIPTION
**Pull request by ivov**
- GET /executions
- GET /executions-current
- GET /executions/:id
- DELETE /executions
- POST /executions/:id/retry
- POST /executions-current/:id/stop

Pinned chokidar to fix build: https://github.com/n8n-io/n8n/pull/2696

### CODE REVIEW

File: packages/cli/src/Server.ts [hunk 1]
```diff
@@ -2731,74 +2735,62 @@ class App {
...
-					});
+					const sharedWorkflowIds = await getSharedWorkflowIds(req.user).then((ids) =>
+						ids.map((id) => id.toString()),
+					);
 
 					for (const data of executingWorkflows) {
 						if (
 							(filter.workflowId !== undefined && filter.workflowId !== data.workflowId) ||
-							workflowsMap[data.workflowId.toString()]
+							!sharedWorkflowIds.includes(data.workflowId)
```
Comment (ivov) [inline packages/cli/src/Server.ts:2789]: Reverted the logic, continue if user does _not_ have access to the workflow.

File: packages/cli/src/WorkflowHelpers.ts [hunk 2]
```diff
@@ -524,3 +527,41 @@ export type NameRequest = Express.Request & {
 		name?: string;
 	};
 };
+
+/**
+ * Build a `where` clause for a `find()` or `findOne()` operation
+ * in the `shared_workflow` or `shared_credentials` tables.
+ */
+export function whereClause({
```
Comment (krynble) [inline packages/cli/src/WorkflowHelpers.ts:535]: I think this function should be moved to `UserManagementHelpers` - we might decide to move them somewhere else later, but I don't think this is the correct place.

File: packages/cli/src/WorkflowHelpers.ts [hunk 3]
```diff
@@ -524,3 +527,41 @@ export type NameRequest = Express.Request & {
...
+		where.user = { id: user.id };
+	}
+
+	return where;
+}
+
+/**
+ * Get the IDs of the workflows that have been shared with the user.
+ */
+export async function getSharedWorkflowIds(user: User): Promise<number[]> {
```
Comment (krynble) [inline packages/cli/src/WorkflowHelpers.ts:557]: Same for this function, I think this should be moved to `UserManagementHelper`

File: packages/cli/src/WorkflowHelpers.ts [hunk 4]
```diff
@@ -524,3 +527,41 @@ export type NameRequest = Express.Request & {
...
+	const sharedWorkflows = await Db.collections.SharedWorkflow!.find({
```
Comment (krynble) [inline packages/cli/src/WorkflowHelpers.ts:558]: We should return all workflow regardless of ownership if the current user is the instance owner.

**Review by krynble — CHANGES_REQUESTED** [review]
Question: any specific reason why we decided to move away from inner joins? These can be a bit faster than 2 queries, but it is no big deal. The changes I made previously already covered some of the changes you did. Was curious about it.

**Comment by ivov** [comment]
> any specific reason why we decided to move away from inner joins?

I raised the same concern about performance, but the decision was to favor readability by abstracting away the SQL.

File: packages/cli/src/WorkflowHelpers.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/WorkflowHelpers.ts:535]: As discussed, to be centralized once other parallel PRs are merged.

File: packages/cli/src/WorkflowHelpers.ts (hunk 3 above)
Comment (ivov) [reply packages/cli/src/WorkflowHelpers.ts:557]: As discussed, to be centralized once other parallel PRs are merged.

File: packages/cli/src/WorkflowHelpers.ts (hunk 4 above)
Comment (ivov) [reply packages/cli/src/WorkflowHelpers.ts:558]: `getSharedWorkflowIds()` calls `whereClause()`, which adds the user ID filter only if the logged-in user is _not_ an owner.

https://github.com/n8n-io/n8n/blob/9ca13617d9aa574b47e0987054d3c16efdd68854/packages/cli/src/WorkflowHelpers.ts#L547-L549

I tested it with the owner and with a member (overriding as below), and it is returning all workflow IDs and the member's workflow IDs, respectively. Are you seeing different behavior? Or would you prefer for the owner's query not to go through `SharedWorkflow` at all, for performance?

```ts
req.user.id = 'someUserId';
req.user.globalRole.name = 'member';
req.user.globalRole.id = 3;
```

Comment (ivov) [reply packages/cli/src/WorkflowHelpers.ts:558]: If the latter, this might be best addressed when creating `UserManagementModel`.

Comment (krynble) [reply packages/cli/src/WorkflowHelpers.ts:557]: You are correct, thanks for pointing it out!

**Review by krynble — APPROVED** [review]

**Pull request merged**